packages/presentation/lib/base/bloc.dart

```diff
@@ -0,0 +1,35 @@
+import 'dart:async';
+import 'package:flutter/material.dart';
+import 'package:get_it/get_it.dart';
+import '../navigation/app_navigator.dart';
+import 'bloc_data.dart';
+
+abstract class Bloc {
+  Stream<BlocData> get dataStream;
```

Comment:
BlocData receives the generic, so u should also pass it here and to all other places where u use BlocData